zkevm-circuits/src/rlp_circuit_fsm.rs

```diff
@@ -275,6 +336,10 @@ pub struct RlpCircuitConfig<F> {
     depth_check: IsEqualConfig<F>,
     /// Check for depth == 1
     depth_eq_one: IsEqualConfig<F>,
+    /// CHeck for depth == 2
+    depth_eq_two: IsEqualConfig<F>,
+    /// Check for depth == 4
+    depth_eq_four: IsEqualConfig<F>,
```

Comment:
Indeed as mentioned, circuit degree will become an issue for adding a 3-bit binary number gadget for depth check. The `tag` related condition requires degree 5, and the `state` related conditions for FSM requires degree 3. Future constraints in this case might have very limited maneuverability as most degrees will be spent on determining the position of the constraint, instead of the actual constraint.